x/staging/subspaces/client/cli/tx.go

```diff
@@ -0,0 +1,249 @@
+			if err != nil {
+				return err
+			}
+
+			subspaceID := args[1]
+
+			msg := types.NewMsgAddAdmin(subspaceID, newAdminAddress.String(), clientCtx.FromAddress.String())
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func GetCmdRemoveSubspaceAdmin() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remove-admin [address] [subspace-id]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Remove an existent admin from the subspace with the given id",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			existentAdminAddress, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			subspaceID := args[1]
+
+			msg := types.NewMsgRemoveAdmin(subspaceID, existentAdminAddress.String(), clientCtx.FromAddress.String())
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func GetCmdRegisterUser() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "register-user [address] [subspace-id]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Register a user inside the subspace with the given [subspace-id] to let him post in it",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			user := args[0]
+			subspaceID := args[1]
+			admin := clientCtx.FromAddress.String()
+			msg := types.NewMsgRegisterUser(user, subspaceID, admin)
+
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func GetCmdBlockUser() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "block-user [address] [subspace-id]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Block a user to post inside the subspace with the given [subspace-id]",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			user := args[0]
+			subspaceID := args[1]
+			admin := clientCtx.FromAddress.String()
+			msg := types.NewMsgBlockUser(user, subspaceID, admin)
+
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func GetCmdEditSubspace() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit [subspace-id] [[new_owner]]",
```

Comment:
Instead of making the `owner` an argument, I would make this a flag with the default value of an empty string (`""`). This would require the user to use it everytime they want to edit a subspace, but I think it would be better than making it an argument.

Reply:
I revised my stance. I think the best way is to set the default value of a flags to `[do-not-modify]` instead to be more consistent with other edit messages (eg. `MsgSaveProfile` or others)